src/pages/TeacherDetails/index.js

```diff
@@ -0,0 +1,33 @@
+import React, { useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+import { fetchTeacherDetails, fetchTeachers } from "../../store/teacher/actions";
+import { fetchSessions } from "../../store/session/actions";
+import Profile from "../../components/Profile";
+import { selectTeacherById } from "../../store/teacher/selectors";
+
+export default function TeacherDetails() {
+
+  const { id } = useParams();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchTeachers());
+  }, [dispatch]);
+
+  useEffect(() => {
+    dispatch(fetchSessions());
+  }, [dispatch]);
```

Comment:
you don't need two useEffects where right? you can do `fetchSessions` and `fetchTeachers` in the same one!